litestar/datastructures/headers.py

```diff
@@ -353,14 +370,22 @@ def prevent_storing(cls) -> "CacheControlHeader":
 
         return cls(no_store=True)
 
+    @classmethod
+    def cast_to_type(cls, field_name: str, value: str) -> Any:
```

Comment:
As mentioned previously, this is broken when `__future__.annotations` is imported, and we are gradually implementing that across the library. Look at `typing_extensions.get_type_hints()`. Although, that something you'd only want to call once per type, not once per access to this method.